core/src/client/importer.rs

```diff
@@ -391,7 +391,9 @@ impl Importer {
                 .block_header(&BlockId::Hash(*header.parent_hash()))
                 .expect("Parent of importing header must exist")
                 .decode();
-            if self.check_header(&header, &parent_header) {
+            if client.block_header(&BlockId::Hash(hash)).is_some() {
```

Comment:
I am worried that this part will be a performance bottleneck because block_header method reads data from disk.

Reply:
Most of the headers accessed by this part will be close to the best block, and they will have been cached with high possibility.
I think we can revisit this if we see a real bottleneck from here.

Reply:
Okay.